garage/tf/algos/npo.py

```diff
@@ -49,7 +50,7 @@ def __init__(self,
         if optimizer is None:
             if optimizer_args is None:
                 optimizer_args = dict()
-            optimizer = LbfgsOptimizer
+            optimizer = PenaltyLbfgsOptimizer
```

Comment:
I think this is an error. True VPG is unconstrainted in KL divergence. Using `PenaltyLbfgsOptimizer` turns it into the soft-penalty version of TRPO.

@amolchanov86 do you agree?

Reply:
you are right

Reply:
This is probably not an error. This is not relevant to VPG because VPG overrides optimizer to FirstOrderOptimizer. I think Chang was to sync up TF with Theano. https://github.com/rlworkgroup/garage/blob/241ae5c02f17e6fb4c6ec59b35454fe4b3dd1c23/garage/theano/algos/npo.py#L26

NPO also uses interface unique of PenaltyLbfgsOptimizer. https://github.com/rlworkgroup/garage/blob/241ae5c02f17e6fb4c6ec59b35454fe4b3dd1c23/garage/theano/algos/npo.py#L93-L99

Reply:
ah i see. so it doesn't matter.

i guess Artem and I just assumed that VPG would be the "default" version of NPO.

Reply:
I guess in the end it doesn't matter because nobody runs NPO directly, and if an algorithm uses a specific optimizer it will set it in the constructor.

I'm ambivalent about whether to revert it or not. Whatever Chang wants.